src/shim/telemetryClient.ts

```diff
@@ -46,14 +58,278 @@ export class TelemetryClient {
+            this._options.otlpTraceExporterConfig = {
+                ...this._options.otlpTraceExporterConfig,
+                keepAlive: !this.config.noHttpAgentKeepAlive,
+            }
+            this._options.otlpMetricExporterConfig = {
+                ...this._options.otlpMetricExporterConfig,
+                keepAlive: !this.config.noHttpAgentKeepAlive,
+            }
+            this._options.otlpLogExporterConfig = {
+                ...this._options.otlpLogExporterConfig,
+                keepAlive: !this.config.noHttpAgentKeepAlive,
+            }
+        }
+
+        if (this.config.ignoreLegacyHeaders === false) {
+            Logger.getInstance().warn("LegacyHeaders are not supported by the shim.");
+        }
+
+        if (this.config.proxyHttpUrl || this.config.proxyHttpsUrl) {
+            const proxyUrl = new URL(this.config.proxyHttpsUrl || this.config.proxyHttpUrl);
+            this._options.azureMonitorExporterConfig.proxyOptions = {
+                host: proxyUrl.hostname,
+                port: Number(proxyUrl.port),
+            }
+        }
+
+        if (this.config.maxBatchSize) {
+            Logger.getInstance().warn("The maxBatchSize configuration option is not supported by the shim.");
+        }
+
+        if (this.config.maxBatchIntervalMs) {
+            this._options.otlpTraceExporterConfig = { ...this._options.otlpTraceExporterConfig, timeoutMillis: this.config.maxBatchIntervalMs };
+            this._options.otlpMetricExporterConfig = { ...this._options.otlpMetricExporterConfig, timeoutMillis: this.config.maxBatchIntervalMs };
+            this._options.otlpLogExporterConfig = { ...this._options.otlpLogExporterConfig, timeoutMillis: this.config.maxBatchIntervalMs };
+        }
+
+        if (this.config.correlationIdRetryIntervalMs) {
+            Logger.getInstance().warn("The correlationIdRetryIntervalMs configuration option is not supported by the shim.");
+        }
+
+        if (this.config.enableLoggerErrorToTrace) {
+            Logger.getInstance().warn("The enableLoggerErrorToTrace configuration option is not supported by the shim.");
+        }
+
+        if (this.config.httpAgent || this.config.httpsAgent) {
+            Logger.getInstance().warn("The httpAgent and httpsAgent configuration options are not supported by the shim.");
+        }
 
+        if (this.config.enableWebInstrumentation || this.config.webInstrumentationConfig || this.config.webInstrumentationSrc || this.config.webInstrumentationConnectionString) {
+            Logger.getInstance().warn("The webInstrumentation configuration options are not supported by the shim.");
+        }
+    }
+    
+    /**
+     * Starts automatic collection of telemetry. Prior to calling start no telemetry will be collected
+     * @param input Set of options to configure the Azure Monitor Client
+     */
+    public start(input?: ApplicationInsightsOptions) {
+        if (_setupCalled) {
```

Comment:
I'm confused why this flag exists, start is not a method that existed before so should be only called in shim-applicationinsights.ts

Reply:
Made some updates to this, but the reasoning is that in the non-shim scenario we want to treat the creation of the `AzureMonitorOTelClient` as part of the constructor of the `TelemetryClient` class, but in the case of the shim we want to separate that logic out between the `setup()` and `start()` methods in the `ApplicationInsights` class. Issue here was that when I refactored, I removed the check from the constructor. Hence why the check didn't make sense. Excellent catch!